### PR TITLE
Apply suggestions from #2

### PR DESCRIPTION
**Pull request by m9brady**
- Use previous day just-in-case current-day data is unavailable
- apply black formatting

Excited to see RIOPS data in Zarr format

### CODE REVIEW

**Comment by andersy005** [comment]
@m9brady, thank you for this contribution. I just submitted a test run to the backend, and once it passes, we'll merge this.

**Comment by m9brady** [comment]
Not sure what I missed... I was able to build the recipe locally and use xarray to load it just fine. Happy to address if the logs show anything.

**Comment by andersy005** [comment]
> Not sure what I missed... I was able to build the recipe locally and use xarray to load it just fine. Happy to address if the logs show anything.

@m9brady, thank you for your patience! there were some issues in backend  environment yesterday. We are currently upgrading the backend environment, and once it's ready and i will trigger new test runs

**Comment by andersy005** [comment]
For reference, this PR was blocked by 
- https://github.com/pangeo-forge/pangeo-forge-recipes/issues/427

File: feedstock/recipe.py [hunk 1]
```diff
@@ -1,43 +1,69 @@
...
 def make_url(variable, time):
+    """Currently hardcoded for 2D variables run at 00Z"""
+    fh = f"{time:03.0f}"
+    # vertical level (SFC or DBS-0.5m for 2D, DBS-all for 3D)
+    if variable in ["VOTEMPER", "VOSALINE", "VOZOCRTX", "VOMECRTY"]:
+        level = "DBS-0.5m"  # Depth-Below-Surface 0.5m
+    else:
+        level = "SFC"  # Surface
+    url = f"{RIOPS_PS_BASE_URL}/2d/00/{fh}/{YYYYMMDD}T00Z_MSC_RIOPS_{variable}_{level}_PS5km_P{fh}.nc"
+    return url
```
Comment (andersy005) [inline feedstock/recipe.py]: @m9brady, apache beam is complaining about the `RIOPS_PS_BASE_URL`. One option is to move this variable inside the `make_url` function body:



```python
  File "/usr/local/lib/python3.9/dist-packages/pangeo_forge_recipes/executors/beam.py", line 40, in exec_stage
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/recipes/xarray_zarr.py", line 155, in cache_input
    fname = config.file_pattern[input_key]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/patterns.py", line 219, in __getitem__
    fname = self.format_function(**format_function_kwargs)
  File "/tmp/tmpmn6sfni2/feedstock/recipe.py", line 23, in make_url
NameError: name 'RIOPS_PS_BASE_URL' is not defined [while running 'Start|cache_input|Reshuffle_000|prepare_target|Reshuffle_001|store_chunk|Reshuffle_002|finalize_target|Reshuffle_003/cache_input/Execute-ptransform-56']
```

Comment (m9brady) [reply feedstock/recipe.py]: Will do - should I do the same for the other globals too? `[START_DATE, YYYYMMDD]`?

Comment (m9brady) [reply feedstock/recipe.py]: Was still able to build and read locally as of https://github.com/pangeo-forge/riops-feedstock/pull/4/commits/20431b2048a861ebb1dbeec12df9181b04bd39df

**Comment by m9brady** [comment]
Apologies for all these failures @andersy005 😭 I appreciate your time and patience

File: feedstock/recipe.py [hunk 2]
```diff
@@ -1,43 +1,70 @@
...
-
-    yyyymmdd = start_date.strftime("%Y%m%d")
-    index = f'{time:03.0f}'
-    return (
+    """Currently hardcoded for 2D variables run at 00Z"""
+    riops_ps_base_url = (
         "https://dd.weather.gc.ca/model_riops/netcdf/forecast/polar_stereographic"
-        f"/2d/00/{index}/{yyyymmdd}T00Z_MSC_RIOPS_{variable}_DBS-0.5m_PS5km_P{index}.nc"
     )
+    start_date = datetime.combine(date.today() - timedelta(days=1), datetime.min.time())
```
Comment (andersy005) [inline feedstock/recipe.py:14]: let's move the `datetime` imports inside the function bodies as well. 

the latest run failed with this error

```python
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/recipes/xarray_zarr.py", line 155, in cache_input
    fname = config.file_pattern[input_key]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/patterns.py", line 219, in __getitem__
    fname = self.format_function(**format_function_kwargs)
  File "/tmp/tmpih5qufhy/feedstock/recipe.py", line 14, in make_url
NameError: name 'datetime' is not defined [while running 'Start|cache_input|Reshuffle_000|prepare_target|Reshuffle_001|store_chunk|Reshuffle_002|finalize_target|Reshuffle_003/cache_input/Execute-ptransform-56']
```

**Comment by andersy005** [comment]
> Apologies for all these failures @andersy005 sob I appreciate your time and patience

no worries @m9brady. happy to help w/ this.  this serialization issue is known and hopefully it will be addressed in the near future 

- https://github.com/pangeo-forge/pangeo-forge-recipes/issues/384#issuecomment-1290450851

File: feedstock/recipe.py (hunk 2 above)
Comment (m9brady) [reply feedstock/recipe.py:14]: I will also move the `from os.path import basename` inside the function where it's used (just-in-case)

Comment (m9brady) [reply feedstock/recipe.py:14]: still happily builds and loads locally as of https://github.com/pangeo-forge/riops-feedstock/pull/4/commits/fd14354891660b7992f48a6a6f3f7c596bb0e894

**Comment by andersy005** [comment]
@m9brady, let me know if this looks good to you. There's an HTML representation of the test run here: https://pangeo-forge.org/dashboard/recipe-run/1343?feedstock_id=5

**Comment by m9brady** [comment]
xc/yc look good but I would expect there to be 84 timesteps (or does the recipe runner use `recipe_pruned`?)

The CRS for the xc/yc can hopefully be fixed in another PR down the road if I can squeeze those details from the data owners. I would also like to see if it's possible to include the other 2D variables (despite being on a different vertical level - surface vs 0.5m below surface). This PR was planned to just fix the broken recipe.

Is that acceptable?

**Comment by andersy005** [comment]
> (or does the recipe runner use recipe_pruned?)

yes, the test run runs a pruned version of the recipe 

> . This PR was planned to just fix the broken recipe. Is that acceptable?

yes. we can definitely address those other issues in a separate PR. 

Welcome to Pangeo-Forge  and thank you very much for your contribution 🚀🎊! Looking forward to more of your contributions :)

**Pull request merged**